lizard_auth_server/views_api_v2.py

```diff
@@ -0,0 +1,227 @@
+"""
+V2 API
+"""
+import logging
+import json
+try:
+    from urlparse import urljoin, urlparse
+    from urllib import urlencode
+except ImportError:
+    from urllib.parse import urljoin, urlparse, urlencode
+
+from django.core.urlresolvers import reverse
+from django.http import HttpResponse
+from django.http import HttpResponseRedirect
+from django.template.context import RequestContext
+from django.template.response import TemplateResponse
+from django.utils.translation import ugettext as _
+import jwt
+
+from lizard_auth_server import forms
+from lizard_auth_server.views_sso import (
+    ProcessGetFormView,
+    domain_match,
+    FormInvalidMixin,
+)
+from lizard_auth_server.views import ErrorMessageResponse
+from lizard_auth_server.models import Profile
+
+
+logger = logging.getLogger(__name__)
+
+
+# Copied from views_sso.py
+def get_domain(form):
+    """Return domain for the redirect back to the site.
+
+    Normally, the ``redirect_url`` is used. If your server is known under
+    several domains, you can pass a ``domain`` GET parameter.
+
+    Note: the domain can also have an extra path element, so
+    http://some.where/something is allowed, if needed.
+
+    """
+    portal_redirect = form.site.redirect_url
+    domain = form.cleaned_data.get('message', {}).get('domain', None)
+
+    # BBB, previously the "next" parameter was used, but django itself also
+    # uses it, leading to conflicts. IF "next" starts with "http", we use it
+    # and otherwise we omit it.
+    next = form.cleaned_data.get('message', {}).get('next', None)
+    if next:
+        if next.startswith('http'):  # Includes https :-)
+            domain = next
+
+    if domain is None:
+        return portal_redirect
+    netloc = urlparse(domain)[1]
+    if netloc == '':
+        return urljoin(portal_redirect, domain)
+    if form.site.allowed_domain != '' \
+            and domain_match(netloc, form.site.allowed_domain):
+        return domain
+    return portal_redirect
+
+
+# Copied from views_sso.py
+def construct_user_data(user=None, profile=None):
```

Comment:
Bij al deze functies even kijken of ze wel nodig zijn. En anders selectief copy/pasten indien je ze nodig hebt.